examples/gameroom/gameroom-app/src/store/api.ts

```diff
@@ -144,43 +175,22 @@ export async function listGames(circuitID: string): Promise<Game[]> {
 
 // Payloads
 export async function submitPayload(payload: Uint8Array): Promise<void> {
-  const options = {
-    method: 'POST',
-    url: `http://${window.location.host}/api/submit`,
-    body: payload,
-    headers: {
-      'Content-Type': 'application/octet-stream',
-    },
-  };
-
-  await rp(options).then((body) => {
-    return;
-  })
-  .catch((err) => {
-    console.error(err.message);
-    throw new Error('Failed to send request. Contact administrator for help.');
+  await http('POST', '/submit', payload, (request: XMLHttpRequest) => {
+    request.setRequestHeader('Content-Type', 'application/octet-stream');
+  }).catch((err) => {
+    throw new Error(err);
   });
 }
 
 export async function submitBatch(payload: Uint8Array, circuitID: string): Promise<BatchInfo[]> {
-  const options = {
-    method: 'POST',
-    url: `http://${window.location.host}/api/gamerooms/${circuitID}/batches`,
-    body: payload,
-    headers: {
-      'Content-Type': 'application/octet-stream',
-    },
-  };
-
-  return await rp(options).then((rawBody) => {
+  return await http('POST', `/gamerooms/${circuitID}/batches`, payload, (request: XMLHttpRequest) => {
+    request.setRequestHeader('Content-Type', 'application/octet-stream');
+  }).catch((err) => {
+    throw new Error(err);
```

Comment:
This is redundant.

Reply:
This is required to send the error back to the component to invoke the toast alert